clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/security/GoogleNamedAccountCredentials.groovy

```diff
@@ -0,0 +1,193 @@
+    List<String> requiredGroupMembership = []
+    String project
+    String applicationName
+    List<String> imageProjects = []
+    ComputeVersion computeVersion = ComputeVersion.V1
+    Map<String, List<String>> regionToZonesMap
+    String jsonKey
+
+    /**
+     * If true, overwrites any value in regionToZoneMap with values from the platform.
+     */
+    boolean regionLookupEnabled = true
+
+    Builder name(String name) {
+      this.name = name
+      return this
+    }
+
+    Builder environment(String environment) {
+      this.environment = environment
+      return this
+    }
+
+    Builder accountType(String accountType) {
+      this.accountType = accountType
+      return this
+    }
+
+    Builder requiredGroupMembership(List<String> requiredGroupMembership) {
+      this.requiredGroupMembership = requiredGroupMembership
+      return this
+    }
+
+    Builder project(String project) {
+      this.project = project
+      return this
+    }
+
+    Builder applicationName(String applicationName) {
+      this.applicationName = applicationName
+      return this
+    }
+
+    Builder imageProjects(List<String> imageProjects) {
+      this.imageProjects = imageProjects
+      return this
+    }
+
+    Builder computeVersion(ComputeVersion version) {
+      this.computeVersion = version
+      return this
+    }
+
+    Builder jsonKey(String jsonKey) {
+      this.jsonKey = jsonKey
+      return this
+    }
+
+    Builder regionToZonesMap(Map<String, List<String>> regionToZonesMap) {
+      this.regionToZonesMap = regionToZonesMap
+      return this
+    }
+
+    Builder regionLookupEnabled(boolean enabled) {
+      this.regionLookupEnabled = enabled
+      return this
+    }
+
+
+    GoogleNamedAccountCredentials build() {
+      GoogleCredentials credentials = jsonKey ?
+          new GoogleJsonCredentials(project, computeVersion, jsonKey) :
+          new GoogleCredentials(project, computeVersion)
+      Compute compute = credentials.getCompute(applicationName)
+
+      if (regionLookupEnabled) {
+        regionToZonesMap = queryRegions(compute, project)
+      }
+
+      new GoogleNamedAccountCredentials(name,
+                                        environment,
+                                        accountType,
+                                        GoogleCloudProvider.GCE,
+                                        requiredGroupMembership,
+                                        credentials,
+                                        project,
+                                        applicationName,
+                                        imageProjects,
+                                        computeVersion,
+                                        regionToZonesMap,
+                                        compute)
+
+    }
+  }
+
+  public List<Map> getRegions() {
+    List<Map> regionList = new ArrayList<Map>()
+
+    for (Map.Entry<String, List<String>> regionToZonesEntry : regionToZonesMap.entrySet()) {
```

Comment:
I think this will NPE if `regionLookupEnabled` is `false` and `regionToZonesMap` is not assigned some value.